kibana-reports/server/routes/utils/visualReportHelper.ts

```diff
@@ -18,85 +18,131 @@ import { FORMAT } from './constants';
-      fullPage: true,
-    });
-
-    //TODO: Add header and footer, phase 2
-
-    await browser.close();
-    return { timeCreated, dataUrl: buffer.toString('base64'), fileName };
+    return renderReport(
+      'image/png',
+      url,
+      source,
+      itemName,
+      windowWidth,
+      windowHeight,
+      header,
+      footer
+    );
   } catch (error) {
     throw error;
   }
 };
 
 export const generatePDF = async (
   url: string,
+  source: string,
   itemName: string,
   windowWidth: number,
-  windowHeight: number
+  windowHeight: number,
+  header: string,
+  footer: string
 ): Promise<{ timeCreated: string; dataUrl: string; fileName: string }> => {
   try {
-    const browser = await puppeteer.launch({
-      headless: true,
-    });
-    const page = await browser.newPage();
+    return renderReport(
+      'application/pdf',
+      url,
+      source,
+      itemName,
+      windowWidth,
+      windowHeight,
+      header,
+      footer
+    );
+  } catch (error) {
+    throw error;
+  }
+};
 
-    await page.setDefaultNavigationTimeout(0);
-    await page.setViewport({
-      width: windowWidth,
-      height: windowHeight,
-    });
+async function renderReport(
+  format: string,
+  url: string,
+  source: string,
+  itemName: string,
+  windowWidth: number,
+  windowHeight: number,
+  header: string,
+  footer: string
+) {
+  if (source === 'Dashboard') {
+    windowWidth = 1440;
+    windowHeight = 2560;
+  } else if (source === 'Visualization') {
+    windowWidth = 900;
+    windowHeight = 1300;
+  }
 
-    await page.goto(url, { waitUntil: 'networkidle0' });
+  const browser = await puppeteer.launch({
+    headless: true,
+  });
+  const page = await browser.newPage();
 
-    const timeCreated = new Date().toISOString();
-    const fileName = getFileName(itemName, timeCreated) + '.' + FORMAT.pdf;
-    // The scrollHeight value is equal to the minimum height the element would require in order to fit
-    // all the content in the viewport without using a vertical scrollbar
-    const scrollHeight = await page.evaluate(
-      () => document.documentElement.scrollHeight
-    );
+  await page.setDefaultNavigationTimeout(0);
+  await page.setViewport({
+    width: windowWidth,
+    height: windowHeight,
+  });
+
+  await page.goto(url, { waitUntil: 'networkidle0' });
+  let element;
+
+  if (source === 'Dashboard') {
+    await page.waitForSelector('#dashboardViewport');
+    element = await page.$('#dashboardViewport');
+  } else if (source === 'Visualization') {
+    await page.waitForSelector('.visChart');
+    element = await page.$('.visChart');
+  }
+
+  const screenshot = await element.screenshot({ fullPage: false });
+  const timeCreated = new Date().toISOString();
 
-    const buffer = await page.pdf({
+  const scrollHeight = await page.evaluate(
+    () => document.documentElement.scrollHeight
+  );
+
+  await page.setContent(`
+  <!DOCTYPE html>
+  <html>
+    <div>
+    <h1>${header}</h1>
```

Comment:
Why is it always `<h1>` ?

Reply:
I thought user can use a markdown editor to pass in anything in the markdown format?

Reply:
`<h1>` is a header placeholder. User can use markdown editor to pass anything but right now we don't have header or footer available in the schema so the hard-coded values are placeholders

Reply:
My understanding is, the client side will convert the markdown header to html elements in string format. And here it will be added inside a `<div>` at the top. So the data schema is actually a string for header/footer. Is that right?

Reply:
The data schema will be a string for header/footer- the parsing into markdown format will likely have to happen server-side before being injected into the screenshot

Reply:
If the goal for this PR is to add a fixed hardcoded header/footer. Then mark important missing features as TODOs in the code, and create issues to track